### PR TITLE
use structured facts

### PR DESCRIPTION
**Pull request by Dan33l**
#### Pull Request (PR) description

move out string facts to use structured facts.

### CODE REVIEW

File: spec/classes/snmp_client_spec.rb [hunk 1]
```diff
@@ -31,11 +30,7 @@
         let(:params) { {} }
         let :facts do
           {
-            osfamily: 'RedHat',
-            operatingsystem: os,
-            operatingsystemrelease: '6.4',
-            lsbmajdistrelease: '6',
-            operatingsystemmajrelease: '6'
+            os: { 'release' => { 'full' => '6.4', 'major' => '6' }, 'name' => 'CentOS', 'family' => 'RedHat' }
```
Comment (alexjfisher) [inline spec/classes/snmp_client_spec.rb]: How easy would it be to start using rspec-puppet-facts?

Comment (Dan33l) [reply spec/classes/snmp_client_spec.rb]: Sure that tests with `rspec-puppet-facts` is better. But, before `4.0.0` is released i wished the module is with structured facts in its code.
If i have enough time before release, i will propose an other one about tests using `rspec-puppet-facts`. It is necessary to clean up some parts. I saw code/tests about old OSes.

For the moment, i propose to merge the PR as this.

Comment (alexjfisher) [reply spec/classes/snmp_client_spec.rb]: As long as the facts you've mocked are accurate and you've not just written them in such a way to make the tests pass! :)

Comment (Dan33l) [reply spec/classes/snmp_client_spec.rb]: Excepted if i made an error, i reported same data but with structured facts instead of string facts

Comment (Dan33l) [reply spec/classes/snmp_client_spec.rb]: Test with `rspec-puppet-facts` will be done in a futur PR.

**Comment by mwhahaha** [comment]
fyi, rspec-puppet-facts doesn't have these facts so it broke folks following master

**Comment by alexjfisher** [comment]
and here https://github.com/camptocamp/facterdb/tree/master/facts for versions available to rspec-puppet-facts

**Pull request merged**